src/main/java/com/gregtechceu/gtceu/api/placeholder/GraphicsComponent.java

```diff
@@ -0,0 +1,57 @@
+package com.gregtechceu.gtceu.api.placeholder;
+
+import com.gregtechceu.gtceu.client.renderer.monitor.IMonitorRenderer;
+import com.gregtechceu.gtceu.common.machine.multiblock.electric.CentralMonitorMachine;
+import com.gregtechceu.gtceu.common.machine.multiblock.electric.monitor.MonitorGroup;
+
+import net.minecraft.client.renderer.MultiBufferSource;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.Tag;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+
+import java.util.function.Supplier;
+
+public record GraphicsComponent(double x, double y, double x2, double y2, String rendererId, CompoundTag renderData)
```

Comment:
Rendering is handled with floats, so there's no use to using doubles here:
```suggestion
public record GraphicsComponent(float x, float y, float x2, float y2, String rendererId, CompoundTag renderData)
```

Reply:
And if you need to have doubles for the constructor, just add a second one that downcasts them to float